src/commands/deploy.ts

```diff
@@ -130,24 +130,31 @@ export async function deploy(ui: IAzureUserInput, actionContext: IActionContext,
-    const children: IAzureNode[] = await node.getCachedChildren();
-    const functionsNode: IAzureParentNode<FunctionsTreeItem> = <IAzureParentNode<FunctionsTreeItem>>children.find((n: IAzureNode) => n.treeItem instanceof FunctionsTreeItem);
-    await node.treeDataProvider.refresh(functionsNode);
-    const functions: IAzureNode[] = await functionsNode.getCachedChildren();
-    const anonFunctions: IAzureNode<FunctionTreeItem>[] = <IAzureNode<FunctionTreeItem>[]>functions.filter((f: IAzureNode) => f.treeItem instanceof FunctionTreeItem && f.treeItem.config.isHttpTrigger && f.treeItem.config.authLevel === HttpAuthLevel.anonymous);
-    if (anonFunctions.length > 0) {
-        ext.outputChannel.appendLine(localize('anonymousFunctionUrls', 'HTTP Trigger Urls:'));
-        for (const func of anonFunctions) {
-            ext.outputChannel.appendLine(`  ${func.treeItem.label}: ${func.treeItem.triggerUrl}`);
+async function listHttpTriggerUrls(node: IAzureParentNode, actionContext: IActionContext): Promise<void> {
+    try {
+        const children: IAzureNode[] = await node.getCachedChildren();
+        const functionsNode: IAzureParentNode<FunctionsTreeItem> = <IAzureParentNode<FunctionsTreeItem>>children.find((n: IAzureNode) => n.treeItem instanceof FunctionsTreeItem);
+        await node.treeDataProvider.refresh(functionsNode);
+        const functions: IAzureNode[] = await functionsNode.getCachedChildren();
+        const anonFunctions: IAzureNode<FunctionTreeItem>[] = <IAzureNode<FunctionTreeItem>[]>functions.filter((f: IAzureNode) => f.treeItem instanceof FunctionTreeItem && f.treeItem.config.isHttpTrigger && f.treeItem.config.authLevel === HttpAuthLevel.anonymous);
+        if (anonFunctions.length > 0) {
+            ext.outputChannel.appendLine(localize('anonymousFunctionUrls', 'HTTP Trigger Urls:'));
+            for (const func of anonFunctions) {
+                ext.outputChannel.appendLine(`  ${func.treeItem.label}: ${func.treeItem.triggerUrl}`);
+            }
         }
-    }
 
-    if (functions.find((f: IAzureNode) => f.treeItem instanceof FunctionTreeItem && f.treeItem.config.isHttpTrigger && f.treeItem.config.authLevel !== HttpAuthLevel.anonymous)) {
-        ext.outputChannel.appendLine(localize('nonAnonymousWarning', 'WARNING: Some http trigger urls cannot be displayed in the output window because they require an authentication token. Instead, you may copy them from the Azure Functions explorer.'));
+        if (functions.find((f: IAzureNode) => f.treeItem instanceof FunctionTreeItem && f.treeItem.config.isHttpTrigger && f.treeItem.config.authLevel !== HttpAuthLevel.anonymous)) {
+            ext.outputChannel.appendLine(localize('nonAnonymousWarning', 'WARNING: Some http trigger urls cannot be displayed in the output window because they require an authentication token. Instead, you may copy them from the Azure Functions explorer.'));
+        }
+    } catch (error) {
```

Comment:
I'd rather see this confined to a smaller area and maybe to a specific error code.

Reply:
Discussed offline. This is confined to the `listHttpTriggerUrls` function, which only occurs after a deployment has already succeeded. No matter the error - we don't want "listing the triggers" to make it seem like the deploy itself has failed